lib/json/stream/buffer.rb

```diff
@@ -29,7 +31,7 @@ def initialize
       def <<(data)
         # Avoid state machine for complete UTF-8.
         if @buffer.empty?
-          data.force_encoding(Encoding::UTF_8)
+          (+data).force_encoding(Encoding::UTF_8)
```

Comment:
Does this work?

```ruby
a = "a"
a.force_encoding("ASCII-8BIT")
a.freeze
(+a).force_encoding(Encoding::UTF_8)
p a.encoding # -> #<Encoding:BINARY (ASCII-8BIT)>
```

Reply:
Sorry it doesn't work. We should assign the result to new variable. I will fix it.
```ruby
a = "a"
a.force_encoding("ASCII-8BIT")
a.freeze
b = (+a).force_encoding(Encoding::UTF_8)

p a.encoding # -> #<Encoding:BINARY (ASCII-8BIT)>
p b.econding # -> #<Encoding:UTF-8>
```

Reply:
`dup` than `+` (conditional `dup`) may be better here because users may not expect that `<<` may change the argument (`data`). 

```ruby
data = data.dup
data.force_encoding(Encoding::UTF_8)
```

Reply:
fix: [d97d4d2](https://github.com/dgraham/json-stream/pull/8/commits/d97d4d243dc5e43566e93aedd3ff204c11a12828) I see. It totally makes sense to me too!